### PR TITLE
Log when you can't mount the app

### PR DESCRIPTION
**Pull request by Stephane-Y**
Sometimes we 404 when requesting the app code. In these cases `constructor = eval.call(window, className);` would throw an exception.

Try and be a bit more descriptive, logging file name of the file that 404'ed

### CODE REVIEW

File: lib/assets/javascripts/react_ujs_mount.js [hunk 1]
```diff
@@ -42,50 +42,27 @@
...
-          if (console && console.log) { console.log("%c[react-rails] %c" + message + " for element", "font-weight: bold", "", node) }
-          var error = new Error(message + ". Make sure your component is globally available to render.")
-          throw error
-        } else {
-          ReactDOM.render(React.createElement(constructor, props), node);
+        var className = node.getAttribute(window.ReactRailsUJS.CLASS_NAME_ATTR);
+        var constructor = window[className];
+
+        if (typeof constructor === "undefined") {
+          if (!props || !props.appFileName) {
```
Comment (dmnd) [inline lib/assets/javascripts/react_ujs_mount.js:59]: `props.appFileName` is a very Flexport-specific thing. At this point I'd probably inline the react-rails code into our repo. It's now pretty tightly coupled to our app. We're never going to merge this stuff upstream.

Comment (Stephane-Y) [reply lib/assets/javascripts/react_ujs_mount.js:59]: I agree with inlining all of this. There is a fair amount of code we don't need from this gem, so we could slim it down a lot.

It's not necessarily a trivial task to inline it though, the gem is responsible for the `react_component` calls in `app.slim.html` and server side rendering, so we need to make sure we keep that around too.

Comment (Stephane-Y) [reply lib/assets/javascripts/react_ujs_mount.js:59]: Hmm 🤔 I might be wrong, we might be able to extract just the `react_ujs` code into our repo and leave the rest of `react-rails` code untouched

File: lib/assets/javascripts/react_ujs_mount.js [hunk 2]
```diff
@@ -42,50 +42,27 @@
...
-      // Fallback to eval to handle cases like 'My.React.ComponentName'.
-      // Also, try to gracefully import Babel 6 style default exports
-      //
-      var constructor;
-
-      // Try to access the class globally first
-      constructor = window[className];
-
-      // If that didn't work, try eval
-      if (!constructor) {
```
Comment (uforic) [inline lib/assets/javascripts/react_ujs_mount.js:57]: Are we sure we want to remove the rest of this getConstructor code?

**Pull request closed without merging**